src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/executeStrategy/noneExecuteStrategy.ts

```diff
@@ -22,13 +23,16 @@ export class NoneExecuteStrategy implements ITerminalExecuteStrategy {
 	readonly type = 'none';
 	private _startMarker: IXtermMarker | undefined;
 
+	private readonly _instance: ITerminalInstance;
+
 	private readonly _onDidCreateStartMarker = new Emitter<IXtermMarker | undefined>;
 	public onDidCreateStartMarker: Event<IXtermMarker | undefined> = this._onDidCreateStartMarker.event;
 
 	constructor(
-		private readonly _instance: ITerminalInstance,
+		private readonly _toolTerminal: IToolTerminal,
```

Comment:
It's probably nicer actually to keep it how it was and still pass in `_instance` but also `_hasReceivedUserInput: () => boolean`. That way the properties on the `IToolTerminal` object can't be messed with here at all (combined with my other comment).